pkgs/by-name/po/poptracker/package.nix

```diff
@@ -0,0 +1,64 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, util-linux
+, SDL2
+, SDL2_ttf
+, SDL2_image
+, openssl
+, which
+, libsForQt5
+, makeWrapper
+}:
+
+stdenv.mkDerivation (finalAttrs: rec {
+  pname = "poptracker";
+  version = "0.25.7";
+
+  src = fetchFromGitHub {
+    owner = "black-sliver";
+    repo = "PopTracker";
+    rev = "v${version}";
+    hash = "sha256-UqGEhs2AY9AKYrItmky3tcvXu/dXnnPwMXKwZ3klXcU=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    SDL2
+    SDL2_ttf
+    SDL2_image
+    openssl
+    util-linux
+    makeWrapper
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    make native CONF=RELEASE
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir $out
+    cp build/linux-x86_64/poptracker $out
+    cp -r assets $out/assets
```

Comment:
```suggestion
    install -m555 -Dt $out/bin build/linux-x86_64/poptracker
    install -m444 -Dt $out/assets assets/*
```

Reply:
you might be able to patch it. grepping around the source i find.
```c
Assets::addSearchPath(os_pathcat(homePopTrackerPath, "assets"));
```
so can probably add a line somewhere in the source like:
```c
Assets::addSearchPath("@assets@");
```
then make a diff and copy it to the same directory as `package.nix` and name it `nix-assets.diff`

then in package.nix do something like.
```nix
{ lib
, other stuff
, substituteAll
}:
[...]
  patches = [
    (substituteAll {
      src = ./nix-assets.diff;
      assets = assets = "${placeholder "out"}/assets"; # maybe use share/assets
    })
  ];
```

and maybe that works

I did something similar in this PR: https://github.com/NixOS/nixpkgs/pull/276408/files

Reply:
Excellent. Poptracker now works from `nix-shell` :+1:

Reply:
hmm -- a couple of things. when running `result/bin/poptracker` it isn't finding the assets. to debug i added.
```nix
  postPatch = ''
    cat src/poptracker.cpp|grep nix/store
    exit 1
  '';
```
which shows
```
patching file src/poptracker.cpp
    Assets::addSearchPath("/nix/store/f8g9rsi5man2cwrxd83l0ivgzvqvxgs8-assets-path.diff/share/assets");
```
so `placeholder "out"` is resolving for for the fixuped patch rather than for where poptracker will get stored. i recall seeing a way around this, but i can't remember exactly what the workaround is. will have to look around.

second, these lines in `installPhase` need to be removed.
```diff
-    mkdir $out
-   cp build/linux-x86_64/poptracker $out
-   cp -r assets $out/assets
```

not quite sure how nix-shell worked to get the assets, perhaps it added a $out to path as well as $out/bin

Reply:
I made a mistake in my testing. It does not actually find the assets when using `nix-shell`. Alternatively I could probably play around with the code to make the binary look in `../assets` relative to itself. I will return to work on this PR later.

Reply:
ok, i asked on the matrix channel and a [work around](https://matrix.to/#/!kjdutkOsheZdjqYmqp:nixos.org/$R4KY1kbT3KRwghwIF5RR53bkH4QegeY1HEI0edxcpl8?via=nixos.org&via=matrix.org&via=nixos.dev) would be to do something like:
https://github.com/NixOS/nixpkgs/blob/c3e128f3c0ecc1fb04aef9f72b3dcc2f6cecf370/pkgs/applications/window-managers/gamescope/default.nix#L60-L64

so patches becomes just `patches = [ ./the.diff ];` and then in postPatch the replace happens.